### ISSUE TITLE
[Feature Request]: transfer table between different DB vendors

### ISSUE THREAD

**Issue by bioschaf** [jlolling/talendcomp_tDBTableTransfer#2]
It would be really nice to be able to use your Talend components to transfer tables between different database servers, especially between different flavours (so e.g. Mysql to MSSQL). The normal way of doing this in Talend is quite cumbersome, as you have to specify the schema for each table.

**Comment by ElRakiti** [comment]
If you have a subscription, you can use the [dynamic schema](https://help.talend.com/reader/9NRa~CylzTWZ8bFvUjlang/yJ3uVEzDciOxdy9B9G1tBw) feature to avoid dealing with fixed schemas.

**Comment by jlolling** [comment]
Hi

I agree with you but the component does more. It converts automatically (or
if needed explicitly) the data types and can handle schema differences -
all these features are not possible with the dynamic schema and of course
for open studio projects there is no other way.

Best regards
Jan

ElRakiti <notifications@github.com> schrieb am Di., 28. Jan. 2020, 09:20:

**Comment by bioschaf** [comment]
I did some testing and looked into the code and configuration files. It is indeed possible to transfer table data from e.g. a MySQL Server to MSSQL. I even tested it with vendors that are not implemented, and was able to use it to transfer data to Sybase with minor code tweaks. Great work!

**Comment by jlolling** [comment]
Hi bioschaf,

I have also some difficulties with Sybase. Could you tell me what you suggest regarding Sybase?
Best regards
Jan

**Comment by bioschaf** [comment]
Hi Jan,
in advance, I was just trying to make it work, so this is by no means elegant. And I only tested this with Sybase 15.
1) Sybase does not seem to support connection.getSchema() So I changed TableTransfer.java to not do that for Sybase-Connections. In my case, it was sufficient to set it to "dbo".
2) Sybase does strange things with catalogs. It seems that .getCatalogs returns about everything on the server, not just in the chosen database. So in SQLDataModel.java I skipped the loadCatalogs() for Sybase-Connections. Not sure if that is OK for all possible use-cases, though.

With this two changes I was able to use your component to transfer data to Sybase. Let me know if I can help.
